lib/modules/versioning/pvp/index.ts

```diff
@@ -0,0 +1,297 @@
+import { logger } from '../../../logger';
+import type { RangeStrategy } from '../../../types/versioning';
+import { regEx } from '../../../util/regex';
+import type { NewValueConfig, VersioningApi } from '../types';
+
+export const id = 'pvp';
+export const displayName = 'Package Versioning Policy (Haskell)';
+export const urls = [];
+export const supportsRanges = true;
+export const supportedRangeStrategies: RangeStrategy[] = ['auto'];
+
+type Parsed = { lower: string; upper: string };
+type Components = { major: number[]; minor: number[]; patch: number[] };
+
+export function parse(input: string): Parsed | null {
+  const noSpaces = input.replaceAll(' ', '');
+  const r = regEx(/>=(?<lower>[\d.]+)&&<(?<upper>[\d.]+)/);
```

Comment:
static regexes should be in the module scope and see my comment about ranges without `&&`